lcls_tools/superconducting/scLinac.py

```diff
@@ -65,29 +65,22 @@ def drivemax(self):
                 else (1 if self.cavity.cryomodule.isHarmonicLinearizer else 0.8))
     
     def calibrate(self, drivemax, attempt=0):
-        print(f"Trying {self.cavity} SSA calibration with drivemax {drivemax}")
-        if drivemax < 0.4:
-            raise utils.SSACalibrationError(f"Requested {self.cavity} SSA drive max too low")
+        print(f"Running {self.cavity} SSA calibration with drivemax {drivemax}")
         
         while caput(self.maxdrive_setpoint_pv, drivemax) != 1:
-            print("Setting max drive")
+            print(f"{self} SSA max drive not set successfully, retrying")
+        
+        if self.cavity.abort_flag:
+            raise utils.CavityAbortError(f"Abort requested for {self.cavity}")
         
         try:
-            if self.cavity.abort_flag:
-                raise utils.CavityAbortError(f"Abort requested for {self.cavity}")
             self.runCalibration()
         
-        except utils.SSACalibrationError as e:
-            print(f"{self.cavity} SSA Calibration failed with '{e}', retrying")
-            self.calibrate(drivemax - 0.02)
-        
-        except utils.SSACalibrationToleranceError as e:
-            print(f"{self.cavity} SSA Calibration failed with '{e}'")
+        except (utils.SSACalibrationToleranceError, utils.SSACalibrationError) as e:
             if attempt < 3:
-                print(f"retyring {self.cavity} calibration")
-                self.calibrate(drivemax, attempt=attempt + 1)
+                self.calibrate(drivemax, attempt + 1)
```

Comment:
why just run it three times with the same parameters?

Reply:
Because sometimes (often) it fails once and then succeeds on a rerun. We ran into this problem last night and it was ROUGH